maestro_python_client/CachedClient.py

```diff
@@ -0,0 +1,95 @@
+from uuid import uuid4
+
+from maestro_python_client.Cache.Cache import Cache
+from maestro_python_client.Client import Client, Task
+
+
+class CachedClient(Client):
+    def __init__(
+        self,
+        maestro_endpoint: str,
+        cache: Cache,
+        cached_queues: list[str] = [],
+        **kwargs,
+    ) -> None:
+        super().__init__(maestro_endpoint, **kwargs)
+        self.__cached_queues: set[str] = set(cached_queues)
+        self.__cache = cache
+
+    def launch_task(
+        self,
+        owner: str,
+        queue: str,
+        task_payload: str,
+        retries: int = 0,
+        timeout: int = 900,
+        executes_in: int = 0,
+    ) -> str:
+        task_payload = self.__cache_payload(
+            queue, task_payload, timeout * (1 + retries) + executes_in
+        )
+        return super().launch_task(
+            owner, queue, task_payload, retries, timeout, executes_in
+        )
+
+    def next(self, queue: str) -> Task | None:
+        task = super().next(queue)
+        return self.__task_from_cache(task)
+
+    def consume(self, queue: str) -> Task | None:
+        task = super().consume(queue)
+        return self.__task_from_cache(task)
+
+    def task_state(self, task_id: str) -> Task:
+        task = super().task_state(task_id)
+        return self.__task_from_cache(task)  # type: ignore
+
+    def complete_task(self, task_id: str, result: str) -> None:
+        task = self.task_state(task_id)
+        result = self.__cache_payload(task.task_queue, result)
+        super().complete_task(task_id, result)
+
+    def launch_task_list(
+        self,
+        tasks: List[Tuple[str, str, str]],
+        retries: int = 0,
+        timeout: int = 900,
+    ) -> List[str]:
+        for i, (owner, queue, payload) in enumerate(tasks):
+            tasks[i] = (
+                owner,
+                queue,
+                self.__cache_payload(queue, payload, timeout * (retries + 1)),
+            )
+
+        return super().launch_task_list(tasks, retries, timeout)
+
+    def __task_from_cache(self, task: Task | None) -> Task | None:
+        if not task:
+            return task
+
+        task.payload = self.__payload_from_cache(task.task_queue, task.payload)
+
+        if task.result:
+            task.result = self.__payload_from_cache(task.task_queue, task.result)
+
+        return task
+
+    def __cache_payload(self, queue: str, payload: str, timeout: int = 0) -> str:
+        if queue not in self.__cached_queues:
+            return payload
+
+        cache_key = self.__unique_key(queue)
+        self.__cache.put(cache_key, payload, timeout)
+        return cache_key
+
+    def __payload_from_cache(self, queue: str, payload: str) -> str:
+        if queue not in self.__cached_queues:
+            return payload
+
+        return self.__cache.get(payload)
+
+    @staticmethod
+    def __unique_key(queue_name: str) -> str:
+        return f"maestro-cache-{queue_name}-{str(uuid4())}"
```

Comment:
Pourquoi ne pas plutôt utiliser le TaskID ? Ça serait plus simple pour débug dans redis le cas échéant.

```suggestion
    @staticmethod
    def __unique_key(queue_name: str, task_id: str, is_payload: bool) -> str:
        return f"maestro-cache-{queue_name}-{task_id}-{'payload' if is_payload else 'result'}"
```

Reply:
Pas possible car on ne connait pas le taskID au moment du `launch_task`